lib/Component.php

```diff
@@ -238,7 +238,13 @@ public function select($name)
                 return array_filter(
                     $result,
                     function ($child) use ($group) {
-                        return $child instanceof Property && strtoupper($child->group) === $group;
+                        if ($child instanceof Property) {
+                            $cgroup = isset($child->group) ? strtoupper($child->group) : '';
+
+                            return $cgroup === $group;
+                        }
+
+                        return false;
```

Comment:
yes, this looks a bit difficult to "engineer" a test to exactly cover this line. Actually the previous check `$child instanceof Property` before `&&` was never covered for the `false` case anyway, but Codecov was not fussy about detecting coverage of each boolean true/false alternative in a logical expression.

So we are really in the same position as before - it is just that Codecov now tells us.

Reply:
This code has been refactored back into a single-line statement.